internal/server/rbac/roles.go

```diff
@@ -742,55 +755,46 @@ func claimListToMap(claims []rbacapi.Claim) map[string][]string {
 	return claimMap
 }
 
-func replaceClaimAnnotations(sa *corev1.ServiceAccount, claims map[string][]string) {
-	// Step through the existing annotations and for each that looks like a claim
-	// annotation, remove it if the corresponding claim is not in the new claims
-	// map.
-	for k := range sa.Annotations {
-		if name, ok := rbacapi.OIDCClaimNameFromAnnotationKey(k); ok {
-			if _, exists := claims[name]; !exists {
-				delete(sa.Annotations, k)
-			}
-		}
-	}
-	if sa.Annotations == nil {
-		sa.Annotations = map[string]string{}
-	}
-	// Add or update the annotations for the new claims
-	for name, values := range claims {
-		sa.Annotations[rbacapi.AnnotationKeyOIDCClaim(name)] = strings.Join(values, ",")
-	}
+func replaceClaimAnnotations(sa *corev1.ServiceAccount, claims map[string][]string) error {
+	return rbacapi.SetOIDCClaimsAnnotation(sa, claims)
```

Comment:
We may as well drop this function and call `rbacapi.SetOIDCClaimsAnnotation()` directly.